src/PRAS2PowerSystems.jl

```diff
@@ -1,240 +1,265 @@
+    resultsspecs = PRASCore.Results.ResultSpec[]
+    push!(resultsspecs, ShortfallSamples())
 
-            csv_path = joinpath(dir_name, string(i), string(asset_dict[k][2], ".csv"))
-            CSV.write(csv_path, df_outage, writeheader=true)
+    for model in template.device_models
+        result_spec = get_pras_resultspec(typeof(get_formulation(model)))
+        if !(result_spec in resultsspecs)
+            push!(resultsspecs, result_spec)
         end
-        @info "Succesfully exported the outage profile for scenario $(i)."
     end
-    @info "Succesfully exported the outage profiles. The CSV files for all asset types for all scenarios are available here : $(dir_name)."
+
+    return resultsspecs
+end
+
+"""
+Get PRAS ResultSpec for RAFormulation
+"""
+function get_pras_resultspec(::Type{GeneratorPRAS})
+    return GeneratorAvailability()
+end
+
+"""
+Get PRAS ResultSpec for RAFormulation
+"""
+function get_pras_resultspec(::Type{B}) where {B <: StoragePRAS}
+    return StorageAvailability()
+end
+
+"""
+Get PRAS ResultSpec for RAFormulation
+"""
+function get_pras_resultspec(::Type{B}) where {B <: GeneratorStoragePRAS}
+    return GeneratorStorageAvailability()
 end
+
+"""
+Get PRAS ResultSpec for AbstractRAFormulation
+"""
+function get_pras_resultspec(::Type{B}) where {B <: AbstractRAFormulation}
+    return error("PRAS ResultSpec not defined for $(B)")
+end
+
+"""
+Get DeviceRAModel for PRAS ResultSpec
+"""
+function get_device_ramodel(::Type{GeneratorAvailability})
+    return GeneratorPRAS
+end
+
+"""
+Get DeviceRAModel for PRAS ResultSpec
+"""
+function get_device_ramodel(::Type{StorageAvailability})
+    return StoragePRAS
+end
+
+"""
+Get DeviceRAModel for PRAS ResultSpec
+"""
+function get_device_ramodel(::Type{GeneratorStorageAvailability})
+    return GeneratorStoragePRAS
+end
+
+"""
+Get DeviceRAModel for PRAS ResultSpec
+"""
+function get_device_ramodel(::Type{R}) where {R <: PRASCore.Results.ResultSpec}
+    return error("DeviceRAModel not defined for $(R)")
+end
+
+"""
+Get AvailabilityResult Type for PRAS ResultSpec
+"""
+function get_pras_resulttype(::Type{GeneratorAvailability})
+    return PRASCore.Results.GeneratorAvailabilityResult
+end
+
+"""
+Get AvailabilityResult Type for PRAS ResultSpec
+"""
+function get_pras_resulttype(::Type{StorageAvailability})
+    return PRASCore.Results.StorageAvailabilityResult
+end
+
+"""
+Get AvailabilityResult Type for PRAS ResultSpec
+"""
+function get_pras_resulttype(::Type{GeneratorStorageAvailability})
+    return PRASCore.Results.GeneratorStorageAvailabilityResult
+end
+
+"""
+Get AvailabilityResult Type for PRAS ResultSpec
+"""
+function get_pras_resulttype(::Type{R}) where {R <: PRASCore.Results.ResultSpec}
+    return error("PRAS AbstractAvailabilityResult not defined for $(R)")
+end
+
+"""
+Get name of Type to strip out parametes for easy comparison
+"""
+name(::Type{T}) where {T} = (isempty(T.parameters) ? T : T.name.wrapper)
```

Comment:
We should rename this to something other than `name`.

Reply:
I renamed it to _type_name_.

Reply:
I removed this in my latest commit because we don't need this anymore.